src/libslic3r/PresetBundle.cpp

```diff
@@ -1533,6 +1533,27 @@ void PresetBundle::load_installed_filaments(AppConfig &config)
                         Preset* filament = filaments.find_preset(filament_iter.first, false, true);
                         if (filament && is_compatible_with_printer(PresetWithVendorProfile(*filament, filament->vendor), PresetWithVendorProfile(printer, printer.vendor)))
                         {
+                            /*
+                            For pellet printers, the flow coefficient is simply converted to a filament diameter that will produce
+                            the same amount of volume. This coefficient is found emperically by testing. 
+
+                            So it is important that the filament_diameter and pellet_flow_coefficient values always stay in sync
+                            */
+                            if (printer.config.opt_bool("pellet_modded_printer")) 
+                            {
+								double pellet_flow_coefficient_value = filament->config.opt_float("pellet_flow_coefficient", 0);
+								double calculated_filament_diamter   = Preset::convert_pellet_flow_to_filament_diameter(
+									pellet_flow_coefficient_value);
+								filament->config.set_key_value("filament_diameter",
```

Comment:
Since there is logic to handle the sync issue in Tab.cpp, is there any reason to have this check here? It's not a desired behavior here